File: android/app/src/main/java/voidawt/image/Raster.java
package voidawt.image;

import voidawt.Point;

public class Raster {
    SampleModel sampleModel;
    DataBuffer dataBuffer;
    int width;
    int height;

    public static WritableRaster createWritableRaster(SampleModel sm, DataBuffer db, Point location) {
        WritableRaster raster = new WritableRaster();
        raster.sampleModel = sm;
        raster.dataBuffer = db;
        raster.width = sm.width;
        raster.height = sm.height;
        return raster;
    }

    public DataBuffer getDataBuffer() {
        return dataBuffer;
    }

    public int getWidth() {
        return width;
    }

    public int getHeight() {
        return height;
    }
}

File: src/commonMain/kotlin/awt/image/Raster.kt
package awt.image

import awt.Point

expect open class Raster {
    fun getNumBands(): Int
    fun getSample(x: Int, y: Int, b: Int): Int
    fun getPixel(x: Int, y: Int, iArray: IntArray?): IntArray
}

expect fun createWritableRaster(sm: SampleModel, db: DataBufferInt, location: Point?): WritableRaster

expect val Raster.pxWidth: Int
expect val Raster.pxHeight: Int

expect val Raster.intData: IntArray

expect fun createPackedRaster(
    dataBuffer: DataBufferInt,
    w: Int,
    h: Int,
    scanlineStride: Int,
    bandMasks: IntArray,
): WritableRaster